### PR TITLE
Accessor methods

### PR DESCRIPTION
**Pull request by vincentneko**
Please let me know if it ok for you before modifying everywhere.

### CODE REVIEW

File: src/Player.ts [hunk 1]
```diff
@@ -73,6 +74,40 @@ export class Player {
...
+
+    public setProduction(resource: Resources, amount : number = 1, game? : Game, fromPlayer? : Player) {
+
+      //Production cannot go negative except MC
+      if (amount < 0) {
+        if ((resource === Resources.STEEL) && (this.steelProduction + amount < 0)) amount = -this.steelProduction;
+        if ((resource === Resources.TITANIUM) && (this.titaniumProduction + amount < 0)) amount = -this.titaniumProduction;
+        if ((resource === Resources.PLANTS) && (this.plantProduction + amount < 0)) amount = -this.plantProduction;
+        if ((resource === Resources.ENERGY) && (this.energyProduction + amount < 0)) amount = -this.energyProduction;
+        if ((resource === Resources.HEAT) && (this.heatProduction + amount < 0)) amount = -this.heatProduction;
```
Comment (bafolts) [inline src/Player.ts]: This is a complex approach to accomplish modifying these properties.

```
if (resource === Resource.MEGACREDITS) this.megaCreditProduction = Math.max(-5, this.megaCreditProduction + amount);
if (resource === Resource.HEAT) this.heatProduction = Math.max(0, this.heatProduction + amount);
```

This should accomplish the same outcome and avoid the complexity above.

**Comment by vincentneko** [comment]
This is it, all resources production have been modified, please review and merge if ok.

**Comment by vincentneko** [comment]
I've mistakenly merged venus-next branch into this one, I need to revert this before merging into master.

**Comment by bafolts** [comment]
You could create a new branch off of `master` and cherry-pick the commits with your changes onto that branch.

```
git checkout -b issue-168
git cherry-pick commit-hash-1
git cherry-pick commit-hash-2
git cherry-pick commit-hash-3
```

File: src/Player.ts [hunk 2]
```diff
@@ -73,6 +75,40 @@ export class Player {
...
+      if (resource === Resources.MEGACREDITS) return this.megaCreditProduction;
+      if (resource === Resources.STEEL) return this.steelProduction;
+      if (resource === Resources.TITANIUM) return this.titaniumProduction;
+      if (resource === Resources.PLANTS) return this.plantProduction;
+      if (resource === Resources.ENERGY) return this.energyProduction;
+      if (resource === Resources.HEAT) return this.heatProduction;
+      return 0;
+    }
+
+    public setProduction(resource: Resources, amount : number = 1, game? : Game, fromPlayer? : Player) {
```
Comment (bafolts) [inline src/Player.ts]: This part relative to the code going to master for the non venus work looks fine. Once you have cherry-picked the commits onto `master` we can merge.

Comment (vincentneko) [reply src/Player.ts]: I don't understand what's happening...
I tried to create a separate branch (revert-merge), from master.
I cherry picked the commits related to the accessors change, but I see I have some venus next code inside as well...

Comment (bafolts) [reply src/Player.ts]: I did the cherry-picking and created a separate pull request.

https://github.com/bafolts/terraforming-mars/pull/176

**Comment by vincentneko** [comment]
Wrong merged with venus next

**Pull request closed without merging**